### PR TITLE
DOC: Updated minor typos in function_base.py and test_function_base.py

### PR DESCRIPTION
**Pull request by madphysicist**
Main correction was to fix up outputs in `nanpercentile` examples and make the docs look more like those of `percentile`.

### CODE REVIEW

**Comment by madphysicist** [comment]
I would like to make the request, if such a thing is possible, to have this PR backported the same way as as https://github.com/numpy/numpy/pull/7180, assuming they are both accepted and 7180 is backported. The reason is that most of this change is to the documentation of `nanpercentile`, so it is very closely related to the code modifications of 7180.

File: numpy/lib/nanfunctions.py [hunk 1]
```diff
@@ -763,20 +763,17 @@ def nanmedian(a, axis=None, out=None, overwrite_input=False, keepdims=False):
...
         type (of the output) will be cast if necessary.
     overwrite_input : bool, optional
-       If True, then allow use of memory of input array (a) for
-       calculations. The input array will be modified by the call to
-       median. This will save memory when you do not need to preserve
-       the contents of the input array. Treat the input as undefined,
-       but it will probably be fully or partially sorted. Default is
-       False. Note that, if `overwrite_input` is True and the input
-       is not already an ndarray, an error will be raised.
+       If True, then allow use of memory of input array (a) for calculations.
```
Comment (jaimefrio) [inline numpy/lib/nanfunctions.py]: Docstrings and comments should wrap at 72 columns, not 79, [PEP 8 says so](https://www.python.org/dev/peps/pep-0008/#maximum-line-length).

Comment (madphysicist) [inline numpy/lib/nanfunctions.py]: Can't argue with PEP8. Fixed. Also, found a couple more typos and changed some wording around while fixing.

File: numpy/lib/function_base.py [hunk 2]
```diff
@@ -3256,15 +3257,14 @@ def median(a, axis=None, out=None, overwrite_input=False, keepdims=False):
...
     Returns
     -------
     median : ndarray
-        A new array holding the result (unless `out` is specified, in which
-        case that array is returned instead).  If the input contains
-        integers, or floats of smaller precision than 64, then the output
-        data-type is float64.  Otherwise, the output data-type is the same
-        as that of the input.
+        A new array holding the result (unless `out` is specified, in
+        which case that array is returned instead).  If the input
```
Comment (charris) [inline numpy/lib/function_base.py]: Could remove `(...)`.

File: numpy/lib/function_base.py [hunk 3]
```diff
@@ -3425,35 +3429,38 @@ def percentile(a, q, axis=None, out=None,
...
-        The percentiles are listed in the first axis. The remaining axes are the
-        reduced axes of the input `a`. If the input contains integers or floats
-        of smaller precision than 64, then the output data-type is float64.
-        Otherwise, the output data-type is the same as that of the input. If
-        `out` is specified, that array is returned instead. 
+        If `q` is a single percentile and `axis=None`, then the result
+        is a scalar. If multiple percentiles are given, the result is an
+        array. The percentiles are listed in the first axis. The
+        remaining axes are the reduced axes of the input `a`. If the
+        input contains integers or floats of smaller precision than 64,
```
Comment (charris) [inline numpy/lib/function_base.py]: The bare `64` is meaningless, maybe `float64` instead.

File: numpy/lib/nanfunctions.py [hunk 4]
```diff
@@ -755,35 +756,34 @@ def nanmedian(a, axis=None, out=None, overwrite_input=False, keepdims=False):
     a : array_like
         Input array or object that can be converted to an array.
     axis : int, optional
-        Axis along which the medians are computed. The default (axis=None)
-        is to compute the median along a flattened version of the array.
-        A sequence of axes is supported since version 1.9.0.
+        Axis along which the medians are computed. The default
+        (axis=None) is to compute the median along a flattened version
```
Comment (charris) [inline numpy/lib/nanfunctions.py]: Hmm. if `None` is the default should have `axis : {int, None}, optional`.  Could then omit `(axis=None)`.

File: numpy/lib/nanfunctions.py [hunk 5]
```diff
@@ -755,35 +756,34 @@ def nanmedian(a, axis=None, out=None, overwrite_input=False, keepdims=False):
...
     out : ndarray, optional
-        Alternative output array in which to place the result. It must have
-        the same shape and buffer length as the expected output, but the
-        type (of the output) will be cast if necessary.
+        Alternative output array in which to place the result. It must
+        have the same shape and buffer length as the expected output,
+        but the type (of the output) will be cast if necessary.
     overwrite_input : bool, optional
-       If True, then allow use of memory of input array (a) for
+       If True, then allow use of memory of input array (`a`) for
```
Comment (charris) [inline numpy/lib/nanfunctions.py]: Remove `()` around `a`

File: numpy/lib/nanfunctions.py [hunk 6]
```diff
@@ -755,35 +756,34 @@ def nanmedian(a, axis=None, out=None, overwrite_input=False, keepdims=False):
...
 
     Returns
     -------
     median : ndarray
-        A new array holding the result. If the input contains integers, or
-        floats of smaller precision than 64, then the output data-type is
-        float64.  Otherwise, the output data-type is the same as that of the
-        input.
+        A new array holding the result. If the input contains integers,
+        or floats of smaller precision than 64, then the output
```
Comment (charris) [inline numpy/lib/nanfunctions.py]: See above about bare `64`.

File: numpy/lib/nanfunctions.py [hunk 7]
```diff
@@ -850,11 +850,13 @@ def nanpercentile(a, q, axis=None, out=None, overwrite_input=False,
     a : array_like
         Input array or object that can be converted to an array.
     q : float in range of [0,100] (or sequence of floats)
-        Percentile to compute which must be between 0 and 100 inclusive.
+        Percentile to compute, which must be between 0 and 100
+        inclusive.
     axis : int or sequence of int, optional
```
Comment (charris) [inline numpy/lib/nanfunctions.py]: `{int, None}`

File: numpy/lib/nanfunctions.py [hunk 8]
```diff
@@ -850,11 +850,13 @@ def nanpercentile(a, q, axis=None, out=None, overwrite_input=False,
...
-        Axis along which the percentiles are computed. The default (None)
-        is to compute the percentiles along a flattened version of the array.
-        A sequence of axes is supported since version 1.9.0.
+        Axis along which the percentiles are computed. The default
+        (None) is to compute the percentiles along a flattened version
```
Comment (charris) [inline numpy/lib/nanfunctions.py]: Omit `(None)`

File: numpy/lib/nanfunctions.py [hunk 9]
```diff
@@ -921,24 +926,21 @@ def nanpercentile(a, q, axis=None, out=None, overwrite_input=False,
     >>> np.nanpercentile(a, 50)
     3.5
     >>> np.nanpercentile(a, 50, axis=0)
-    array([[ 6.5,  4.5,  2.5]])
-    >>> np.nanpercentile(a, 50, axis=1)
+    array([ 6.5,  2.,  2.5])
+    >>> np.nanpercentile(a, 50, axis=1, keepdims=True)
```
Comment (charris) [inline numpy/lib/nanfunctions.py]: This is correct?

File: numpy/lib/nanfunctions.py [hunk 10]
```diff
@@ -921,24 +926,21 @@ def nanpercentile(a, q, axis=None, out=None, overwrite_input=False,
...
     array([[ 7.],
            [ 2.]])
     >>> m = np.nanpercentile(a, 50, axis=0)
     >>> out = np.zeros_like(m)
     >>> np.nanpercentile(a, 50, axis=0, out=m)
-    array([[ 6.5,  4.5,  2.5]])
+    array([[ 6.5,  2.,  2.5]])
     >>> m
-    array([[ 6.5,  4.5,  2.5]])
```
Comment (charris) [inline numpy/lib/nanfunctions.py]: Why change?

File: numpy/lib/nanfunctions.py (hunk 9 above)
Comment (madphysicist) [inline numpy/lib/nanfunctions.py]: Yes, because the nan in the middle is omitted.

Comment (madphysicist) [inline numpy/lib/nanfunctions.py]: Yes. Any percentile of a single number is that number.

File: numpy/lib/nanfunctions.py (hunk 10 above)
Comment (madphysicist) [inline numpy/lib/nanfunctions.py]: Missed that one. Thanks for the catch. Fixed now.

**Pull request merged**